### PR TITLE
[SCM-925] Implement RemoveCommand in maven-scm-provider-jgit

### PR DESCRIPTION
**Pull request by G-Ork**
Thrird try to fix [SCM-925]
- Implement remove command for JGit provider
- Write TCK for remove command.
- Changed JGIT from 4.5.4.201711221230-r to 5.13.0.202109080827-r
- Changed Java from 1.7 to 1.8 (JGit moves forward)
- Change codehaus java checks  from 1.7 to 1.8

### CODE REVIEW

**Comment by michael-o** [comment]
Massive, thanks. Will take  a look during Christmas

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml [hunk 1]
```diff
@@ -88,4 +95,11 @@
       </plugin>
     </plugins>
   </build>
+  
+  <properties>
+  	<!-- jgit version: 6.0.0.202111291000-r   need java 11
+         jgit version: 5.13.0.202109080827-r  need java  8  
+  	-->
+  	<org.eclipse.jgit.version>5.13.0.202109080827-r</org.eclipse.jgit.version>
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:103]: Since this requires Java 8, it won't happen in 1.x. Can this be done with an older version?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml [hunk 2]
```diff
@@ -50,9 +50,16 @@
...
       <groupId>org.eclipse.jgit</groupId>
       <artifactId>org.eclipse.jgit</artifactId>
-      <version>4.5.4.201711221230-r</version>
+      <version>${org.eclipse.jgit.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>org.eclipse.jgit</groupId>
+      <artifactId>org.eclipse.jgit.ssh.jsch</artifactId>
+      <version>${org.eclipse.jgit.version}</version>
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:59]: Why is this necessary? JGit has moved to MINA.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml (hunk 1 above)
Comment (G-Ork) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:103]: It is not neccesary. I just tried to upgrade some very old dependencies. Git is evolving, JGIT shurely fixes a lot of bugs. Maven also moved to Java 8. It is paradox to fix something and keep downward compatibility while everyone moving forward. 

Are there plans to upgrade the plugins to 3.x?

But my heart whont bleed if keeping the old dependency.

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml (hunk 2 above)
Comment (G-Ork) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:59]: Just try to maintain and upgrade an old dependency. Is there any problem with apache MINA?

Comment (michael-o) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:59]: No problem with MINA, as far as I understood the JSch approach is dead.

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml (hunk 1 above)
Comment (michael-o) [reply maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/pom.xml:103]: The 1.x line will remain on Java 7. 2.x will move to Java 8. So please make a decision how imporant this is for you then I can continue to review your PR.

**Comment by natami** [comment]
Any progress on this change? :)

**Comment by michael-o** [comment]
@natami I am waiting for @G-Ork to move unrelated changes to separate PRs.

**Comment by G-Ork** [comment]
Closed and create a new #144 containing only the bare minimum.

**Comment by jira-importer** [comment]
Resolve #1150
